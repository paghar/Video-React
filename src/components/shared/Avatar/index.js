//@flow
import React from 'react';
import { AvatarBtn } from './index.style';
import Typography from 'components/shared/Typography';

type Props = {
  backgroundColor: string,
  imgUrl?: string,
  isBorder?: boolean,
  children: string,
  width: string,
  height: string,
  borderRadius: string,
  font: Object,
  color: string,
  avatarClick?: () => void,
  className?: string,
};

function Avatar({
  backgroundColor,
  imgUrl,
  isBorder,
  children,
  width,
  height,
  borderRadius,
  font,
  color,
  avatarClick,
  className,
}: Props) {
  return (
    <AvatarBtn
      backgroundColor={backgroundColor}
      imgUrl={imgUrl}
      isBorder={isBorder}
      width={width}
      height={height}
      borderRadius={borderRadius}
      onClick={avatarClick}
      className={className}
    >
      {!imgUrl && (
        <Typography font={font} color={color} hoverable>
          {children}
        </Typography>
      )}
    </AvatarBtn>
  );
}

export default Avatar;
