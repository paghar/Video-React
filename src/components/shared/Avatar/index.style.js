//@flow
import styled from 'styled-components';

export const AvatarBtn = styled.button`
  background-color:${({ backgroundColor }) => backgroundColor}  
  ${({ imgUrl }) => imgUrl && `background-image: url(${imgUrl});`}
  background-size: 100% 100%;
   width: ${({ width }) => width};
  height: ${({ height }) => height};
  border-radius: ${({ borderRadius }) => borderRadius};
  border: ${({ isBorder }) => (isBorder ? '2px' : '0px')} solid #0d0d1a;  
  display: flex;
  justify-content: center;
  align-items: center;
  outline: none;
`;
