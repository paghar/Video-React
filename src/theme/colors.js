import theme from 'styled-theming';

export const Generals = {
  i: theme('mode', {
    dark: '#05050d',
  }),
  ii: theme('mode', {
    dark: '#0d0d1a',
  }),
  iii: theme('mode', {
    dark: '#151526',
  }),
  iv: theme('mode', {
    dark: '#212133',
  }),
  v: theme('mode', {
    dark: '#262640',
  }),
  vi: theme('mode', {
    dark: '#7b7b9a',
  }),
  vii: theme('mode', {
    dark: '#b7b7cc',
  }),
  viii: theme('mode', {
    dark: '#ffffff',
  }),
  x: theme('mode', {
    dark: '#e6e6f3',
  }),
  ix: theme('mode', {
    dark: '#f1f1f9',
  }),
};

export const Primary = {
  Dim: theme('mode', {
    dark: '#2172ff',
  }),
  Light: theme('mode', {
    dark: '#317BFC',
  }),
  Dark: theme('mode', {
    dark: '#0E5EE8',
  }),
};

export const PromptWarn = {
  Dim: theme('mode', {
    dark: '#F51653',
  }),
  Light: theme('mode', {
    dark: '#FF4074',
  }),
  Dark: theme('mode', {
    dark: '#DE0540',
  }),
};

export const Secondary = {
  i: theme('mode', {
    dark: '#00d3b2',
  }),
  ii: theme('mode', {
    dark: '#ffcc00',
  }),
  iii: theme('mode', {
    dark: '#f51653',
  }),
  iv: theme('mode', {
    dark: '#64a7ff',
  }),
};

export const Avatars = [
  theme('mode', {
    dark: '#5c4dff',
  }),
  theme('mode', {
    dark: '#685ce6',
  }),
  theme('mode', {
    dark: '#7d73e6',
  }),
  theme('mode', {
    dark: '#928ae6',
  }),
];

export const Gradient = theme('mode', {
  dark:
    'linear-gradient(to bottom, rgba(21, 21, 38, 0), rgba(21, 21, 38, 0.69) 53%, #111120)',
});
