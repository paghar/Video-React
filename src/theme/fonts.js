import theme from 'styled-theming';

export const Title1 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '22px',
  }),
  fontWeight: theme('size', {
    medium: '400',
  }),
  lineHeight: theme('size', {
    medium: '35px',
  }),
};

export const Title2 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '21px',
  }),
  fontWeight: theme('size', {
    medium: '500',
  }),
  lineHeight: theme('size', {
    medium: '33px',
  }),
};

export const Display1 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '14px',
  }),
  fontWeight: theme('size', {
    medium: '400',
  }),
  lineHeight: theme('size', {
    medium: '23px',
  }),
};

export const Display1M = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '14px',
  }),
  fontWeight: theme('size', {
    medium: '500',
  }),
  lineHeight: theme('size', {
    medium: '23px',
  }),
};

export const Display2 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '13px',
  }),
  fontWeight: theme('size', {
    medium: '400',
  }),
  lineHeight: theme('size', {
    medium: '21px',
  }),
};
export const Display3 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '13px',
  }),
  fontWeight: theme('size', {
    medium: '500',
  }),
  lineHeight: theme('size', {
    medium: '21px',
  }),
};
export const Display4 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '12px',
  }),
  fontWeight: theme('size', {
    medium: '400',
  }),
  lineHeight: theme('size', {
    medium: '20px',
  }),
};
export const Display5 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '11px',
  }),
  fontWeight: theme('size', {
    medium: '400',
  }),
  lineHeight: theme('size', {
    medium: '18px',
  }),
};
export const Display6 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '11px',
  }),
  fontWeight: theme('size', {
    medium: '500',
  }),
  lineHeight: theme('size', {
    medium: '18px',
  }),
};
