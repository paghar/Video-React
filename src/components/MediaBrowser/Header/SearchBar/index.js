// @flow
import React, { useState } from 'react';
import Input from 'components/shared/Input';
import { Display3 } from 'theme/fonts';
import { Generals } from 'theme/colors';

type Props = {
  className?: string,
};

const SearchBar = ({ className }: Props) => {
  const [searchIcon, setSearchIcon] = useState('61442');
  const [searchValue, setSearchValue] = useState('');
  const onInputChanged = event => {
    const {
      target: { value: inputValue },
    } = event;
    setSearchValue(inputValue);
    if (inputValue) {
      setSearchIcon('61453');
    } else {
      setSearchIcon('61442');
    }
  };
  const purgeSearch = () => {
    setSearchIcon('61442');
    setSearchValue('');
  };

  return (
    <Input
      height="31px"
      width="223px"
      font={Display3}
      borderRadius="40px"
      textColor={Generals.vii}
      backgroundColor={Generals.iv}
      hoverBackgroundColor={Generals.iv}
      placeholder="Search here..."
      placeHolderColor={Generals.vi}
      icon={searchIcon}
      iconColor={Generals.vii}
      iconSize="15px"
      paddingRL="14px"
      className={className}
      changed={onInputChanged}
      iconClicked={purgeSearch}
      value={searchValue}
    />
  );
};

export default SearchBar;
