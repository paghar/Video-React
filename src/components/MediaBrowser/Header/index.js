// @flow
import React from 'react';
import {
  Wrapper,
  SearchInput,
  AddUserButton,
  NotificationsButton,
  NewButton,
  ShareButton,
} from './index.style';
import Typography from 'components/shared/Typography';
import { Title1, Display1 } from 'theme/fonts';
import { Generals, Primary } from 'theme/colors';
import ProjectPersons from 'components/shared//ProjectPersons';
import type { AvatarSection } from 'components/shared//ProjectPersons';

type Props = {
  projectTitle: string,
  avatarsClicked: () => void,
  addUserClicked: () => void,
  shareClicked: () => void,
  newClicked: () => void,
  notificationClicked: () => void,
  className?: string,
};

const Header = ({
  projectTitle,
  fiveProjectUsers,
  otherUsersCount,
  avatarsClicked,
  addUserClicked,
  shareClicked,
  newClicked,
  notificationClicked,
  className,
}: Props & AvatarSection) => {
  return (
    <Wrapper className={className}>
      <Typography font={Title1} color={Generals.viii} hoverable>
        {projectTitle}
      </Typography>
      <SearchInput />
      <AddUserButton
        rtl
        backgroundColor={Generals.v}
        hoverBackgroundColor={Generals.iv}
        activeBackgroundColor={Generals.iv}
        radius="50%"
        paddingRight="0px"
        icon="61447"
        iconSize="13px"
        iconColor={Generals.vii}
        width="31px"
        height="31px"
        clicked={addUserClicked}
      />
      <ProjectPersons
        fiveProjectUsers={fiveProjectUsers}
        otherUsersCount={otherUsersCount}
        clicked={avatarsClicked}
      />

      <ShareButton
        backgroundColor={Primary.Dim}
        hoverBackgroundColor={Primary.Light}
        radius="5px"
        icon="61920"
        iconColor={Generals.viii}
        spacing="4.8px"
        width="75px"
        iconSize="14px"
        height="34px"
        activeBackgroundColor={Primary.Dark}
        clicked={shareClicked}
      >
        <Typography font={Display1} color={Generals.viii}>
          Share
        </Typography>
      </ShareButton>
      <NewButton
        backgroundColor={Generals.v}
        hoverBackgroundColor={Generals.iv}
        radius="5px"
        iconSize="13px"
        icon="61543"
        spacing="4.8px"
        width="63px"
        height="34px"
        clicked={newClicked}
        iconColor={Generals.viii}
        activeBackgroundColor={Generals.iv}
      >
        <Typography font={Display1} color={Generals.viii}>
          New
        </Typography>
      </NewButton>
      <NotificationsButton
        rtl
        backgroundColor="transparent"
        hoverBackgroundColor={Generals.iv}
        activeBackgroundColor={Generals.v}
        radius="50%"
        paddingRight="0px"
        icon="61683"
        iconSize="15px"
        iconColor={Generals.vii}
        width="31px"
        height="31px"
        clicked={notificationClicked}
      />
    </Wrapper>
  );
};

export default Header;
