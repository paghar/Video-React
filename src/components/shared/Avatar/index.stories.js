import React from 'react';
import { action } from '@storybook/addon-actions';
import Story from '.';
import { Display3 } from 'theme/fonts';
import image from 'assets/image/christopher-campbell-rDEOVtE7vOs-unsplash.png';

export default { title: 'Avatar' };

export const Avatar = () => (
  <Story
    backgroundColor="#5c4dff"
    imgUrl={image}
    width="31px"
    height="31px"
    borderRadius="100%"
    font={Display3}
    color="#ffffff"
    avatarClick={action('avatarClick')}
  >
    PZ
  </Story>
);

export const NoimageAvatar = () => (
  <Story
    backgroundColor="#5c4dff"
    width="33px"
    height="33px"
    borderRadius="100%"
    font={Display3}
    color="#ffffff"
  >
    PZ
  </Story>
);

export const WithBorder = () => (
  <Story
    backgroundColor="#5c4dff"
    isBorder
    width="31px"
    height="31px"
    borderRadius="100%"
    font={Display3}
    color="#ffffff"
  >
    PZ
  </Story>
);

export const WithImageBorder = () => (
  <Story
    backgroundColor="#5c4dff"
    imgUrl={image}
    isBorder
    width="31px"
    height="31px"
    borderRadius="100%"
    font={Display3}
    color="#ffffff"
  >
    PZ
  </Story>
);

export const differentSize = () => (
  <Story
    backgroundColor="#5c4dff"
    isBorder
    width="100px"
    height="100px"
    borderRadius="100%"
    font={Display3}
    color="#ffffff"
  >
    PZ
  </Story>
);
