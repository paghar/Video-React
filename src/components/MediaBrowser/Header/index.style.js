import styled from 'styled-components';
import Button from 'components/shared/Button';
import SearchBar from './SearchBar';

export const Wrapper = styled.div`
  width: 100%;
  display: flex;
  align-items: center;
`;

export const SearchInput = styled(SearchBar)`
  margin-left: auto;
`;

export const AddUserButton = styled(Button)`
  margin-left: auto;
  margin-right: 8px;
`;

export const ShareButton = styled(Button)`
  margin-left: 11px;
`;

export const NewButton = styled(Button)`
  margin-left: 7px;
`;

export const NotificationsButton = styled(Button)`
  margin-left: 4px;
`;
