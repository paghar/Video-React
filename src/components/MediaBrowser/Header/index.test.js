//@flow
import React from 'react';
import renderer from 'react-test-renderer';
import { Display6 } from 'theme/fonts';
import Header from '.';
import { shallow, mount } from 'enzyme';
import { Avatars as AvatarColors } from 'theme/colors';
import image from 'assets/image/christopher-campbell-rDEOVtE7vOs-unsplash.png';
import Avatar from 'components/shared/Avatar';

describe('<MediaBrowserHeader />', () => {
  it('Header Snapshot test', () => {
    const tree = renderer
      .create(
        <Header
          projectTitle="Animal House"
          fiveProjectUsers={[
            {
              color: AvatarColors[0],
              name: 'PZ',
              image,
            },
            {
              color: AvatarColors[0],
              name: 'PZ',
              image,
            },
          ]}
          otherUsersCount={3}
          avatarsClicked={console.log}
          addUserClicked={console.log}
          shareClicked={console.log}
          newClicked={console.log}
          notificationClicked={console.log}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Header Titles check', () => {
    const wrapper = shallow(
      <Header
        projectTitle="Animal House"
        fiveProjectUsers={[
          {
            color: AvatarColors[0],
            name: 'PZ',
            image,
          },
          {
            color: AvatarColors[0],
            name: 'PZ',
            image,
          },
        ]}
        otherUsersCount={3}
        avatarsClicked={console.log}
        addUserClicked={console.log}
        shareClicked={console.log}
        newClicked={console.log}
        notificationClicked={console.log}
      />
    );

    expect(wrapper.html().includes('Animal House')).toEqual(true);
  });
  it('Header events test', () => {
    const mockAvatarsClick = jest.fn();
    const mockAddUserClick = jest.fn();
    const mockShareClick = jest.fn();
    const mockNewClick = jest.fn();
    const mockNotificationClick = jest.fn();
    const wrapper = mount(
      <Header
        projectTitle="Animal House"
        fiveProjectUsers={[
          {
            color: AvatarColors[0],
            name: 'PZ',
            image,
          },
          {
            color: AvatarColors[0],
            name: 'PZ',
            image,
          },
        ]}
        otherUsersCount={3}
        avatarsClicked={mockAvatarsClick}
        addUserClicked={mockAddUserClick}
        shareClicked={mockShareClick}
        newClicked={mockNewClick}
        notificationClicked={mockNotificationClick}
      />
    );
    wrapper
      .find(Avatar)
      .at(0)
      .simulate('click');
    expect(mockAvatarsClick).toHaveBeenCalled();
    wrapper
      .find('button')
      .at(0)
      .simulate('click');
    expect(mockAddUserClick).toHaveBeenCalled();
    wrapper
      .find('button')
      .at(4)
      .simulate('click');
    expect(mockShareClick).toHaveBeenCalled();
    wrapper
      .find('button')
      .at(5)
      .simulate('click');
    expect(mockNewClick).toHaveBeenCalled();
    wrapper
      .find('button')
      .at(6)
      .simulate('click');
    expect(mockNotificationClick).toHaveBeenCalled();
  });
});
