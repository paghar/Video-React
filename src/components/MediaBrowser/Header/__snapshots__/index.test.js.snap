// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaBrowserHeader /> Header Snapshot test 1`] = `
.c5 {
  font-family: 'FontAwesome' !important;
  font-size: 15px;
  margin: 0px;
  padding: 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c9 {
  font-family: 'FontAwesome' !important;
  font-size: 13px;
  margin: 0px;
  padding: 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c16 {
  font-family: 'FontAwesome' !important;
  margin: 0px;
  padding: 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c21 {
  font-family: 'FontAwesome' !important;
  font-size: 14px;
  margin: 0px;
  padding: 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c10 {
  font-size: 13px;
  display: block;
  padding-right: 0px;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c17 {
  display: none;
  padding-right: 0px;
  padding-left: 0px;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c22 {
  font-size: 14px;
  display: block;
  padding-right: 4.8px;
  padding-left: 0px;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c25 {
  font-size: 13px;
  display: block;
  padding-right: 4.8px;
  padding-left: 0px;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c28 {
  font-size: 15px;
  display: block;
  padding-right: 0px;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c7 {
  direction: rtl;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  background: none;
  width: 31px;
  height: 31px;
  border-radius: 50%;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c14 {
  direction: ltr;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  background: none;
  width: 33px;
  height: 33px;
  border-radius: 100%;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c19 {
  direction: ltr;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  background: none;
  width: 75px;
  height: 34px;
  border-radius: 5px;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c23 {
  direction: ltr;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  background: none;
  width: 63px;
  height: 34px;
  border-radius: 5px;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c26 {
  direction: rtl;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  background: none;
  background-color: transparent;
  width: 31px;
  height: 31px;
  border-radius: 50%;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c4 {
  outline: none;
  border: none;
  background: none;
  width: 100%;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c4 {
  outline: none;
  border: none;
  background: none;
  width: 100%;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c4 {
  outline: none;
  border: none;
  background: none;
  width: 100%;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c4 {
  outline: none;
  border: none;
  background: none;
  width: 100%;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c6 {
  padding-right: 0px;
  padding-left: 14px;
  font-size: 15px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c2 {
  box-sizing: border-box;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  direction: ltr;
  border-radius: 40px;
  width: 223px;
  height: 31px;
  padding-right: 14px;
  padding-left: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  margin-left: auto;
}

.c8 {
  margin-left: auto;
  margin-right: 8px;
}

.c20 {
  margin-left: 11px;
}

.c24 {
  margin-left: 7px;
}

.c27 {
  margin-left: 4px;
}

.c1 {
  pointer-events: none;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c18 {
  pointer-events: auto;
}

.c12 {
  background-color: background-image:url(christopher-campbell-rDEOVtE7vOs-unsplash.png);
  background-size: 100% 100%;
  width: 33px;
  height: 33px;
  border-radius: 100%;
  border: 2px solid #0d0d1a;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c13 {
  margin-left: -8px;
}

.c13:first-child {
  margin-left: 0;
}

.c15 {
  border: 2px solid #000;
  margin-left: -8px;
}

<div
  className="c0"
>
  <span
    className="c1"
  >
    Animal House
  </span>
  <div
    className="c2 c3"
    height="31px"
    width="223px"
  >
    <input
      className="c4"
      onChange={[Function]}
      placeholder="Search here..."
      value=""
    />
    <span
      className="c5 c6"
      fontSize="15px"
      onClick={[Function]}
    >
      
    </span>
  </div>
  <button
    className="c7 c8"
    height="31px"
    onClick={[Function]}
    radius="50%"
    width="31px"
  >
    <span
      className="c9 c10"
      fontSize="13px"
    >
      
    </span>
  </button>
  <div
    className="c11"
    onClick={[Function]}
  >
    <button
      className="c12 c13"
      height="33px"
      width="33px"
    />
    <button
      className="c12 c13"
      height="33px"
      width="33px"
    />
    <button
      className="c14 c15"
      height="33px"
      radius="100%"
      width="33px"
    >
      <span
        className="c16 c17"
      >
         
      </span>
      <span
        className="c18"
      >
        +3
      </span>
    </button>
  </div>
  <button
    className="c19 c20"
    height="34px"
    onClick={[Function]}
    radius="5px"
    width="75px"
  >
    <span
      className="c21 c22"
      fontSize="14px"
    >
      
    </span>
    <span
      className="c18"
    >
      Share
    </span>
  </button>
  <button
    className="c23 c24"
    height="34px"
    onClick={[Function]}
    radius="5px"
    width="63px"
  >
    <span
      className="c9 c25"
      fontSize="13px"
    >
      
    </span>
    <span
      className="c18"
    >
      New
    </span>
  </button>
  <button
    className="c26 c27"
    height="31px"
    onClick={[Function]}
    radius="50%"
    width="31px"
  >
    <span
      className="c5 c28"
      fontSize="15px"
    >
      
    </span>
  </button>
</div>
`;
